pkg/libursa/ursa/ursa.go

```diff
@@ -119,6 +119,83 @@ func CredentialPublicKeyFromJSON(jsn string) (unsafe.Pointer, error) {
 	return handle, nil
 }
 
+func CorrectnessProofToJSON(credSignatureCorrectnessProof unsafe.Pointer) ([]byte, error) {
+	var proofOut *C.char
+
+	result := C.ursa_cl_signature_correctness_proof_to_json(credSignatureCorrectnessProof, &proofOut)
+	defer C.free(unsafe.Pointer(proofOut))
+	defer C.free(credSignatureCorrectnessProof)
+	if result.code != 0 {
```

Comment:
Since credSignatureCorrectnessProof was passed in, you shouldn't be freeing it.  The caller of this method may want to do something with it after turning to JSON.